src/components/ChannelList/hooks/listeners/useChannelUpdated.js

```diff
@@ -0,0 +1,28 @@
+import { useContext, useEffect } from 'react';
+
+import { ChatContext } from '../../../../context';
+
+export const useChannelUpdated = ({ onChannelUpdated, setChannels }) => {
+  const { client } = useContext(ChatContext);
+
+  useEffect(() => {
+    const handleEvent = (e) => {
+      if (onChannelUpdated && typeof onChannelUpdated === 'function') {
+        onChannelUpdated(setChannels, e);
+      } else {
+        setChannels((channels) => {
+          const index = channels.findIndex(
+            (c) => c.cid === (e.cid || e.channel?.cid),
+          );
+          if (index >= 0 && e.channel) {
+            channels[index].data = e.channel;
+          }
+          return [...channels];
+        });
+      }
+    };
+
+    client.on('channel.updated', handleEvent);
+    return () => client.off('channel.updated', handleEvent);
```

Comment:
something we should bring up to the team for future changes to the js lib as I've seen it in some other libraries. having the return of the `client.on` function be the `'off'` function. so then all we'd need to do here would be `const channelUpdatedEventHandler = client.on('channel.updated', handleEvent);` and then `return channelUpdatedEventHandler;`

Reply:
We already do that!! e.g., 

```js
const channelUpdatedEventHandler = client.on('channel.updated', handleEvent);

// While unmounting
channelUpdatedEventHandler.unsubscribe();
```

Reply:
sweet